docker-compose.yml

```diff
@@ -0,0 +1,31 @@
+# Configuration for the services that are required to run an analysis of the
+# data from riff edu mattermost.
+# see https://www.metabase.com/docs/latest/operations-guide/running-metabase-on-docker.html
+version: '3.6'
+services:
+
+  target-db:
+    image: mysql:5.7
+    container_name: mysql-business-analytics
+    environment:
+      MYSQL_ROOT_PASSWORD: root
+    ports:
+      - '3306:3306'
+    volumes:
+      - target-db-data:/var/lib/mysql
+
+  metabase:
+    image: metabase/metabase:latest
+    container_name: metabase-business-analytics
+    environment:
+      MB_DB_FILE: ./metabase-db/metabase.db
+    ports:
+      - '3000:3000'
+    volumes:
+      - ./metabase-db:/metabase-db
+    depends_on:
+      - target-db
+
+volumes:
+  target-db-data:
+  metabase-db:
```

Comment:
similarly this one might be `ba-metabase-db`, just adding the prefix to help when looking at the lists of docker volumes.
Although now that I look more closely I see this volume is not actually referenced by any service.

Reply:
Ah you're right. I guess since I'm just binding the local folder to the folder inside the metabase container, a volume is never created?

Reply:
Right.
Now whether you should be binding to a local folder is a different question.

That will depend on the instructions for getting this up and running.
Ideally, you can export the questions/dasboard into some file that the setup instructions will explain how to import, so you can use a named volume and not have to put the
metabase database into the repo. I'm not sure where you've gotten with that.

Reply:
It isn't clear to me why that is ideal. What are the downsides to having the metabase database in the repo?

Reply:
It's opaque, it changes when used, so you don't know if you need to commit it again, it can't be merged (because it's an opaque binary file).

Reply:
Ah I see. Yes, it does change. But unless you change a question or anything on the dashboard, you shouldn't commit it.

Reply:
I still think it's a better option than having to setup and configure the entire metabase app including user info, database configurations, the dashboard, and each question in the dashboard (and configuring the dynamic variables in each question).

Reply:
It's possible the dynamic variables get output in the question export. This might be something I should test locally now. It still sounds like a lot of work to just get something up and running locally.

Reply:
The available options for export/import may indeed be more trouble than they're worth at this time, it would be better for many reasons, but may not be worth the effort, the question is what is the effort. And the available options may have addition downsides, again just questions to be answered in order to come to a reasoned decision.

Reply:
Fair enough. Alright I will look into question export options and get back to you later today.

Reply:
+1 on not putting the metabase DB in the repo. 

Relatedly, does the metabase DB contain PII? I think it does. In that case, it is also a bad practice for us to store it unencrypted on local machines.

Reply:
It doesn't store anything from the mattermost db. Just the sql queries (questions), dashboard, and general configuration of metabase.

Reply:
Ah, okay. I'm less concerned about it then.

Reply:
It is admittedly a little awkward to have checked in, partially due to the opaqueness of the code. As you have this software running locally, the metabase DB file will change. But I recommend not committing it unless you make a change to the metabase config, questions, dashboard layouts, etc. This definitely seems easier and more straightforward to me than having to configure metabase every time you run this software. This would involve setting up a user account, configuring the database connection, importing each question and configuring its dynamic variables, and possibly organising a dashboard for ease of use. Having to do that each time seems inefficient. 

As a second note, it is recommended by the metabase community, when moving to a production environment, to abandon the use of the H2 database and migrate to a postgres/mysql database to store the metabase database.

Reply:
That's something to look into going forward. Configure metabase to use the mysql database we're running anyway, and commit the backup of that database instead of the binary database file.
The backup of a mysql database is a text file, and therefore can be diffed, and possibly merged, although that is still likely to be difficult.
There may be reasons that won't work, but it's worth keeping in mind and exploring at some point.